src/Io/ChunkedDecoder.php

```diff
@@ -122,12 +122,22 @@ public function handleData($data)
                     }
                 }
 
-                $this->chunkSize = \hexdec($hexValue);
-                if (\dechex($this->chunkSize) !== $hexValue) {
+                if (
+                    !(
+                        \PHP_VERSION_ID >= 70400 &&
+                        (
+                            (\function_exists('ctype_xdigit') && \ctype_xdigit($hexValue) === false) ||
+                            // From: https://github.com/symfony/polyfill-ctype/blob/f8f0b461be3385e56d6de3dbb5a0df24c0c275e3/Ctype.php#L196
+                            (\is_string($hexValue) && '' !== $hexValue && !preg_match('/[^A-Fa-f0-9]/', $hexValue))
+                        )
+                    ) ||
+                    \dechex(\hexdec($hexValue)) !== $hexValue
```

Comment:
That's a lot of code :eyes: 

What do you think about this https://3v4l.org/pSZCT? PHP's documentation suggests it should *ignore* all invalid characters, which matches the output, so we may as well ignore the deprecation warning (which would only affect PHP 7.4+)?

Reply:
@clue here is the issue: https://3v4l.org/Hiqht

Reply:
@clue together with: https://github.com/reactphp/http/blob/master/tests/Io/ChunkedDecoderTest.php#L297

Reply:
@clue could still do `@\dechex(@\hexdec($hexValue)) !== $hexValue` to validate it tho

Reply:
@clue just pushed a simpler version, which falls back to the `@\dechex(@\hexdec($hexValue)) !== $hexValue` one fails (because the maxlength one was failing while the rest passed)